integration-cli/docker_cli_daemon_test.go

```diff
@@ -2174,27 +2174,33 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *check.C) {
 
 	infoLog := "\x1b[34mINFO\x1b"
 
+	b := bytes.NewBuffer(nil)
+
 	p, tty, err := pty.Open()
 	c.Assert(err, checker.IsNil)
-	defer func() {
-		tty.Close()
-		p.Close()
-	}()
-
-	b := bytes.NewBuffer(nil)
 	go io.Copy(b, p)
 
 	// Enable coloring explicitly
 	s.d.StartWithLogFile(tty, "--raw-logs=false")
 	s.d.Stop(c)
 	c.Assert(b.String(), checker.Contains, infoLog)
 
+	// "tty" is closed in s.d.Stop(), we need open another pair of pty
+	tty.Close()
```

Comment:
Why the double close of `tty` if it is closed in `Stop()`. I think defer before protected tty being closed in case of early failure.